entrypoint.sh

```diff
@@ -11,15 +11,25 @@ fi
 config=`echo $configs | head -n 1`
 interface="${config%.*}"
 
+if [[ "$(cat /proc/sys/net/ipv4/conf/all/src_valid_mark)" != "1" ]]; then
+    echo "sysctl net.ipv4.conf.all.src_valid_mark=1 is not set" >&2
+    exit 1
+fi
+
+sed -i "s:sysctl -q net.ipv4.conf.all.src_valid_mark=1:echo skipping setting net.ipv4.conf.all.src_valid_mark:" /usr/bin/wg-quick
 wg-quick up $interface
 
 docker_network="$(ip -o addr show dev eth0 | awk '$3 == "inet" {print $4}')"
 docker_network_rule=$([ ! -z "$docker_network" ] && echo "! -d $docker_network" || echo "")
 iptables -I OUTPUT ! -o $interface -m mark ! --mark $(wg show $interface fwmark) -m addrtype ! --dst-type LOCAL $docker_network_rule -j REJECT
 
 docker6_network="$(ip -o addr show dev eth0 | awk '$3 == "inet6" {print $4}')"
-docker6_network_rule=$([ ! -z "$docker6_network" ] && echo "! -d $docker6_network" || echo "")
-ip6tables -I OUTPUT ! -o $interface -m mark ! --mark $(wg show $interface fwmark) -m addrtype ! --dst-type LOCAL $docker6_network_rule -j REJECT
+if [[ -z "${docker6_network}" ]]; then
+    echo "skipping ipv6 killswitch setup since ipv6 interface was not found..." >&2
+else
+    docker6_network_rule=$([ ! -z "$docker6_network" ] && echo "! -d $docker6_network" || echo "")
+    ip6tables -I OUTPUT ! -o $interface -m mark ! --mark $(wg show $interface fwmark) -m addrtype ! --dst-type LOCAL $docker6_network_rule -j REJECT
+fi
```

Comment:
Could you elaborate on why this change was required? It's a solid change, but I don't quite follow how it's related to removing `--privileged`

Reply:
Sorry for the confusion, this is a separate change from the privileged. I was having issues starting the container due to not having a docker ipv6 network setup, so I added this to prevent setting up ipv6 rules if there isn't an ipv6 interface.

Reply:
Ah, fair enough! In that case, would you mind making this a separate PR?

Reply:
Sure, removed it from this PR and opened a new PR: https://github.com/jordanpotter/docker-wireguard/pull/4

Reply:
Thank you so much!